Added orcaflex export function

RubendeBruin opened the pull request:
Adds a function to export a single spectrum to the [orcaflex](https://www.orcina.com/orcaflex/) time-domain simulation program.

Note that an orcaflex license is required to do this.
The tests will try to run this test but will not fail if the orcaflex API can not be loaded.

Running the tests on a machine with orcaflex license gives the following output:

Time-domain signal from orcaflex simultation:
![image](https://user-images.githubusercontent.com/34062862/114504950-38e21200-9c62-11eb-92f3-56463995e166.png)

Comparison oned() input spectrum and spectrum re-calculated from time-domain signal using welch() 👍 
![image](https://user-images.githubusercontent.com/34062862/114505024-57480d80-9c62-11eb-8eb4-cbc19206b38f.png)

Screenshot of orcaflex model with exported spectrum:
![image](https://user-images.githubusercontent.com/34062862/114505064-65962980-9c62-11eb-85ac-00ce94605c73.png)

rafa-guedes commented:
Thanks @RubendeBruin that will be useful addition to orcaflex users.

A few comments:

- It would be good dealing with the cases where the SpecDataset will be composed by multiple spectra - either by allowing looping through non-spectral dimensions, or by raising an exception when that happens. If you decide to go for the former, as a suggestion I have just written an [export function to funwave spectra](https://github.com/wavespectra/wavespectra/pull/36/files#diff-e27d527c65e85a23edd6f6e2f7825d6bd12f46e96ff7949624f65051130c7a4e) (not merged yet) that loops through all spectra in dataset, if more than one spectrum, and writes then into a zip archive.
- I don't know much about the software, does it have a specific file format for ingesting spectra and would it perhaps make sense to write that model instance as such a file?
- Is there any published documentation describing this spectra processing / ingestion that you could point to in the docstring?
- Could you please run [black](https://pypi.org/project/black/) on the code as this is the standard used for the rest of the code base.

RubendeBruin commented:
Hi @rafa-guedes 

Thanks for your review.

- Orcaflex is a time-domain simulation. It will only accept a single environmental condition for a simulation. It has its own datafile format which contains all the data needed to run the simulation. As such the wave-spectrum it uses is embedded in the orcaflex datafile.
The export function in this pull request uses the orcaflex API to inject a single spectrum into an existing model. Exporting more that a single spectrum is not possible using this approach (we would have to deep-copy the whole model, and then export save each one of them, but that feels like a very non-typical usecase).

- I didn't know about black. Think it is great :-)
- Added a link to the documentation to the docstring.
- Added check one single spectrum and tests on that check

RubendeBruin commented:
PLEASE HOLD

there are some work-arounds faster exporting of large amounts of data that I'd like to add to the commit.
